x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/compare_synced_integrations.ts

```diff
@@ -0,0 +1,326 @@
+      : undefined;
+    const result: RemoteSyncedCustomAssetsRecord = {};
+
+    // compare custom pipelines and custom component templates
+    Object.entries(ccrCustomAssets).forEach(([ccrCustomName, ccrCustomAsset]) => {
+      const res = compareCustomAssets({
+        ccrCustomAsset,
+        ingestPipelines: installedPipelines,
+        componentTemplates,
+      });
+      result[ccrCustomName] = res;
+    });
+    return result;
+  } catch (error) {
+    logger.error('error', error?.message);
+    return { error: error?.message };
+  }
+};
+
+const compareCustomAssets = ({
+  ccrCustomAsset,
+  ingestPipelines,
+  componentTemplates,
+}: {
+  ccrCustomAsset: CustomAssetsData;
+  ingestPipelines?: IngestGetPipelineResponse;
+  componentTemplates?: Record<string, ClusterComponentTemplateSummary>;
+}): RemoteSyncedCustomAssetsStatus => {
+  const result = {
+    name: ccrCustomAsset.name,
+    type: ccrCustomAsset.type,
+    package_name: ccrCustomAsset.package_name,
+    package_version: ccrCustomAsset.package_version,
+  };
+
+  if (ccrCustomAsset.type === 'ingest_pipeline') {
+    if (!ingestPipelines) {
+      if (ccrCustomAsset.is_deleted === true) {
+        return {
+          ...result,
+          sync_status: 'completed' as SyncStatus.COMPLETED,
+        };
+      }
+      return {
+        ...result,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+
+    const installedPipeline = ingestPipelines[ccrCustomAsset?.name];
+    if (ccrCustomAsset.is_deleted === true && installedPipeline) {
+      return {
+        ...result,
+        sync_status: 'failed' as SyncStatus.FAILED,
+        error: `Asset ${ccrCustomAsset.name} marked for deletion found installed`,
+      };
+    } else if (isEqual(installedPipeline, ccrCustomAsset?.pipeline)) {
+      return {
+        ...result,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+      };
+    } else if (
+      installedPipeline?.version &&
+      installedPipeline.version < ccrCustomAsset.pipeline.version
+    ) {
+      return {
+        ...result,
+        sync_status: 'failed' as SyncStatus.FAILED,
+        error: `Found incorrect installed version ${installedPipeline.version}`,
+      };
+    } else {
+      return {
+        ...result,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+  } else if (ccrCustomAsset.type === 'component_template') {
+    if (!componentTemplates) {
+      if (ccrCustomAsset.is_deleted === true) {
+        return {
+          ...result,
+          sync_status: 'completed' as SyncStatus.COMPLETED,
+        };
+      }
+      return {
+        ...result,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+
+    const installedCompTemplate = componentTemplates[ccrCustomAsset?.name];
+    if (ccrCustomAsset.is_deleted === true && installedCompTemplate) {
+      return {
+        ...result,
+        sync_status: 'failed' as SyncStatus.FAILED,
```

Comment:
same here, should these be synchronizing unless we know there was an error deleting?

Reply:
It's handled below: https://github.com/elastic/kibana/pull/216178/files#diff-f8de6a6d308d65b9d61c400a2fdbebe1078e67eaf036537c028476260be254f5R312-R315

I left the `throw` block and handled outside because we might need this function elsewhere, this is a basic utility for the ccr case.